packages/dara-core/js/shared/utils/use-component-registry.tsx

```diff
@@ -14,27 +16,31 @@ interface ComponentRegistryInterface {
  */
 function useComponentRegistry(maxRetries = 5): ComponentRegistryInterface {
     const { componentRegistry: components, refetchComponents } = useContext(RegistriesCtx);
-
+    const token = useSessionToken();
     const get = useCallback(
         async (instance: ComponentInstance): Promise<Component> => {
             let component: Component = null;
             let registry = { ...components };
-
             let i = 0;
-            while (i < maxRetries) {
-                if (registry && registry[instance.name]) {
-                    component = registry[instance.name];
-                    break;
-                } else {
-                    // If component has not been found, it could be a nested py_component, so we refetch the registry
-                    // to see if the component might have been added to the registry in the meantime
-                    // But first wait for 0,5s before retrying, to give time for backend to update the registry
-                    await new Promise((resolve) => setTimeout(resolve, 500));
-                    const { data } = await refetchComponents();
-                    registry = data;
+            if (registry && registry[instance.name]) {
+                return registry[instance.name];
+            }
+            try{
+                const res = await request(`/api/core/components/${instance.name}`, { method: HTTP_METHOD.GET }, token);
+                return await res.json();
+            }catch{
+                while (i < maxRetries) {
+                        // If component has not been found, it could be a nested py_component, so we refetch the registry
+                        // to see if the component might have been added to the registry in the meantime
+                        // But first wait for 0,5s before retrying, to give time for backend to update the registry
+                        await new Promise((resolve) => setTimeout(resolve, 500));
+                        const { data } = await refetchComponents();
+                        registry = data;
+                        if (registry && registry[instance.name]) {
+                            return registry[instance.name];
+                    }
+                    i++;
                 }
-
-                i++;
             }
```

Comment:
I think the logic here could be simplified a little bit, now you have duplication where you have this bit

```
if (registry && registry[instance.name]) {
   return registry[instance.name];
```

twice, once before the loop and once within. Could we contain all logic within the loop, and let's add some comments. The rough flow here should be:
- For each iteration (until max retries)
1) check if value is already in registry, return if that's the case and quit the loop
2) otherwise make a direct request for the component (invoking custom lookup potentially)
3) refetch the registry regardless of whether 2) succeeded, just to keep our 
  local registry copy fresh to prevent future requests
4) if we found a component in 2), return it